addon.gypi

```diff
@@ -138,7 +138,11 @@
       [ 'OS in "freebsd openbsd netbsd solaris" or \
          (OS=="linux" and target_arch!="ia32")', {
         'cflags': [ '-fPIC' ],
-      }]
+      }],
+      [ 'OS == "android"', {
+        'cflags': [ '-fPIC' ],
+        'cflags_cc': ['-fPIC' ],
+      }],
```

Comment:
You can fold this into the previous condition: `OS in "android freebsd ..."`.  You don't need to add it to `cflags_cc` because `cflags` is for both C and C++ code.

Reply:
@bnoordhuis thanks for your comment. Your proposal was my first attemp to fix it, but it didn't. To verify if I didn't make any mistake I just did some testing:
- with 'cflags_cc': ['-fPIC' ] is compiles
- without 'cflags_cc': ['-fPIC' ] is does not
I think I have to compare the verbose outputs in more detail to figure out what happens. Any ideas?

Reply:
`env V=1 node-gyp rebuild` turns on verbose compilation mode (i.e., prints the compile commands.)

Reply:
This is the compiler command for the variant **without ccflags_cc**. It seems that the `-fPIE` after the first `-fPIC` creates and unlinkable shared library on android.
`...  -fPIC -Wall -Wextra -Wno-unused-parameter -O3 -fno-omit-frame-pointer -fPIE  -fno-rtti ...`

**With ccflags_cc** the command is build like that. It seems, that the later `-fPIC` overrides the preceding `-fPIE` so that it produces the right shared library.
`...  -fPIC -Wall -Wextra -Wno-unused-parameter -O3 -fno-omit-frame-pointer -fPIE -fPIC -fno-rtti ...`

With this result I would rather try to remove the `-fPIE` while compiling shared libraries on android. I added an additional entry in `target_condition` for removing `-fPIE` like shown below. That did the trick and produces the following compiler command.
`... -fPIC -Wall -Wextra -Wno-unused-parameter -O3 -fno-omit-frame-pointer -fno-rtti ...`

Excerpt of my local `addon.gypi`:
```
'target_conditions': [
...
      ['library=="shared_library"', {
        'conditions': [
          [ 'OS=="android"', {
            'cflags!': [ '-fPIE' ],
          }]
        ]
      }],
...
]
```
@bnoordhuis, what is your comment on this?

Reply:
Ah, you're probably getting hit by #1118 - that `-fPIE` flag is not set by node-gyp, it's inherited from node's common.gypi.

The right thing to do until #1118 is resolved is probably to suppress the flag in node-gyp's addon.gypi.

It must have been broken for quite a while because that flag was added 20 months ago in https://github.com/nodejs/node/pull/5544.